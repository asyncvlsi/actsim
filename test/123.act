namespace pfloat {
    export
    template<pint EXP,MAN>
    deftype float(int<1> sign; int<EXP> exponent; int<MAN> significand) {
        methods {
            function is_infinite() : bool {
                chp {
                    self := exponent = (1 << EXP) - 1 & significand = 0
                }
            }

            function is_finite() : bool {
                chp {
                    // self := ~(exponent = (1 << EXP) - 1 & significand != 0)
                    self := is_infinite()
                }
            }
        }
    }

    export
    defproc test() {
        float<1,5> y;
        chp {
            y := float<1,5>(0);
            [ y.is_finite() -> log("here ")
           [] else -> log ("there ")
            ]
        }
    }
}


defproc test()
{
   pfloat::test t;
}
