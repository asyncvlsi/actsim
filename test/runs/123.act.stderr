WARNING: test<>: substituting chp model (requested prs, not found)
